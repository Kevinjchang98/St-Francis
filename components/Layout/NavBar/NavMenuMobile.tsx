'use client';
import Image from 'next/image';
import { useState } from 'react';
import styles from './NavBar.module.css';
import type { MenuItems } from './NavMenu';
import NavMenu from './NavMenu';

export type NavMenuMobileProps = {
    items?: MenuItems;
    icon?: string;
};

export default function NavMenuMobile(props: NavMenuMobileProps) {
    const { items = [], icon = '/menu.svg' } = props;
    const [show, setShow] = useState(false);

    const handleShow = () => {
        window.scrollTo(0, 0);
        document.documentElement.style.overflow = show ? 'scroll' : 'hidden';
        setShow(!show);
    };

    return show ? (
        <div className={styles.navMobile} onClick={handleShow}>
            <NavMenu className={styles.navItemsMobile} items={items} />
        </div>
    ) : (
        <button className={styles.navMenuButton} onClick={handleShow}>
            <Image src={icon} alt="menu" width="20" height="20" />
        </button>
    );
}
