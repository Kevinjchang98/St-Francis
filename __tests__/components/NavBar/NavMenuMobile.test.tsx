import NavMenuMobile from '@/components/Layout/NavBar/NavMenuMobile';
import '@testing-library/jest-dom';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

global.scrollTo = jest.fn();

describe('NavMenuMobile', () => {
    it('matches snapshot without any props', async () => {
        const tree = renderer.create(<NavMenuMobile />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders without any props', async () => {
        const { container } = render(<NavMenuMobile />);
        expect(container).toBeInTheDocument();
    })

    it('shows menu when button is clicked', async () => {
        const { container, getByRole } = render(<NavMenuMobile />);
        const button = getByRole('button');
        fireEvent.click(button);
        expect(container).toBeInTheDocument();
    })
});
