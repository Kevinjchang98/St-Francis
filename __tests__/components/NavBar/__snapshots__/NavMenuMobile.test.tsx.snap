// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenuMobile matches snapshot without any props 1`] = `
<button
  className="navMenuButton"
  onClick={[Function]}
>
  <img
    alt="menu"
    data-nimg="1"
    decoding="async"
    height={20}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/menu.svg"
    style={
      {
        "color": "transparent",
      }
    }
    width={20}
  />
</button>
`;
